sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubClientFactory.cs

```diff
@@ -106,7 +110,7 @@ internal EventProcessorHost GetEventProcessorHost(string eventHubName, string co
                 }
 
                 return new EventProcessorHost(consumerGroup: consumerGroup,
-                    connectionString: NormalizeConnectionString(info.ConnectionString, eventHubName),
+                    connectionString: info.ConnectionString,
```

Comment:
Not quite.  The Event Hubs clients do not allow you to specify a connection string with a different Event Hub name than the one that you passed in explicitly.   If both are present, it will assert they're the same and throw if they are not.

Because the connection string AND event hub name were passed in here, they could not disagree.